src/rgw/rgw_compression_types.h

```diff
@@ -44,25 +44,33 @@ WRITE_CLASS_ENCODER(compression_block)
 struct RGWCompressionInfo {
   string compression_type;
   uint64_t orig_size;
+  boost::optional<int32_t> compressor_message;
   vector<compression_block> blocks;
 
   RGWCompressionInfo() : compression_type("none"), orig_size(0) {}
   RGWCompressionInfo(const RGWCompressionInfo& cs_info) : compression_type(cs_info.compression_type),
                                                           orig_size(cs_info.orig_size),
+							  compressor_message(cs_info.compressor_message),
                                                           blocks(cs_info.blocks) {}
 
   void encode(bufferlist& bl) const {
-    ENCODE_START(1, 1, bl);
+    ENCODE_START(2, 1, bl);
     encode(compression_type, bl);
     encode(orig_size, bl);
+    if(struct_v >= 2) {
```

Comment:
you probably don't need this check for encoding, new version is always applied

Reply:
If we upgrade from an old version to a new version, we need to make this check in order to ensure that the original compressed data can be decompress

Reply:
this does apply to decompress but makes no sense for encode which is always performed under the latest version.